### ISSUE TITLE
Project 3 Demo

### ISSUE THREAD

**Issue by jakepruitt** [SreeniASU/multimedia-information-systems#41]
As of right now, these are the available times:

| Date | Day | time |
| --- | --- | --- |
| 11/2/2015 | Wednesday | 01:00 PM - 02:00 PM |
| 11/2/2015 | Wednesday | 02:15 PM - 03:15 PM |
| 11/2/2015 | Wednesday | 03:30 PM - 04:30 PM |

What can everyone do? I'm afraid I am leaving town on Wednesday at 1:30, so unless we can trade with another group, I won't be able to make this demo. If anyone wants to sit down over the next two days and run through the demo, I'm free tomorrow evening to discuss how everything works.

cc/ @Perryo @ApertureTS @SreeniASU @GCrispino

**Comment by ApertureTS** [comment]
I have a class that ends at 1, so the last 2 times would work for me.
Will you be on campus tomorrow or meet

**Comment by Perryo** [comment]
@jakepruitt 
I'm assuming this is for 12/2/2015, I can make any of those times, just let me know so i can put it in my phone.

**Comment by SreeniASU** [comment]
Anytime works with me for 12/02/2015

On Mon, Nov 30, 2015 at 6:05 PM, James notifications@github.com wrote:

> @jakepruitt https://github.com/jakepruitt
> I'm assuming this is for 12/2/2015, I can make any of those times, just
> let me know so i can put it in my phone.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/SreeniASU/multimedia-information-systems/issues/41#issuecomment-160813719
> .

## 

Sreenivas P Shenoy
M.S. Computer Science & Engineering
Arizona State University,
spsheno1@asu.edu
Cell 480-738-3280

**Comment by GCrispino** [comment]
I have a class that finishes at 1:15 PM on Wednesday. The second time is best for me, but I can be there at the third one too.

**Comment by jakepruitt** [comment]
Great, signed up for the 2:15-3:15 block!

**Comment by Perryo** [comment]
@jakepruitt 
Looked over dwt, I think i've got a good grasp on it based on what i've read in the code/report. One question however. After the high/low pass filters and array concatenation, the values are passed through the filters again to create the 4 quadrant transform. Why is this done again, and are you specifically slicing the array to create these quadrants?

**Comment by jakepruitt** [comment]
So the LL quadrant (top-right hand corner after first iteration of DWT) has the most information out of the four quadrants produced in the first iteration. If we can take this information and distill it down **even further** we can get really good compression and isolate the most significant aspects in the image.

After the first iteration is done, we perform a second iteration on that top-left (in this case 4x4) quadrant, and then a third iteration on the top-left quadrant (2x2) of that. This results in the top left number containing more information than all of the other numbers - it effectively is a summary of the entire 8x8 region and is equivalent to taking the average of all of the pixel Y values (another way of phrasing this is that it is the 1x1 resolution version of the 8x8 block).

This picture does a better job than my words:

<img width="183" alt="screenshot 2015-12-01 19 15 26" src="https://cloud.githubusercontent.com/assets/5084263/11520341/f1ebcec4-985f-11e5-83b4-2a94972807c5.png">

If the professor or TA ask you "Why did you do it that much? Why not stop at one?" The answer you should say is "The requirements asked us for the _most significant_ wavelet components. Repeating DWT condenses the most important information down into one pixel, and following the zig-zag ordering on those wavelets will produce the most important information. If you were to only have the first n most significant wavelets, you would still be able to reproduce the image pretty well."

Does that answer your question?

**Comment by Perryo** [comment]
@jakepruitt 
Perfect, exactly what I was wondering. I'm trying to understand everything so Yash can't get off on trying to confuse us out of points.

**Comment by SreeniASU** [comment]
Yash has mailed to use values of n and m. But how is value of 100 possible
for DCT or DWT. We have only 64 values per frame block. :-/

On Tue, Dec 1, 2015 at 8:07 PM, James notifications@github.com wrote:

> @jakepruitt https://github.com/jakepruitt
> Perfect, exactly what I was wondering. I'm trying to understand everything
> so Yash can't get off on trying to confuse us out of points.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/SreeniASU/multimedia-information-systems/issues/41#issuecomment-161165796
> .

## 

Sreenivas P Shenoy
M.S. Computer Science & Engineering
Arizona State University,
spsheno1@asu.edu
Cell 480-738-3280

**Comment by Perryo** [comment]
@SreeniASU 
Maybe its him trying to check the outputs and see if it behaves weirdly?

**Comment by SreeniASU** [comment]
I have mailed him. Waiting for the reply :)

On Tue, Dec 1, 2015 at 10:37 PM, James notifications@github.com wrote:

> @SreeniASU https://github.com/SreeniASU
> Maybe its him trying to check the outputs and see if it behaves weirdly?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/SreeniASU/multimedia-information-systems/issues/41#issuecomment-161186733
> .

## 

Sreenivas P Shenoy
M.S. Computer Science & Engineering
Arizona State University,
spsheno1@asu.edu
Cell 480-738-3280

**Comment by GCrispino** [comment]
Hey guys. Sorry for getting late here, but did some of you see that Yash put the demo videos and that we have to process it and bring the generated features to the video?

**Comment by ApertureTS** [comment]
Do you want me to print the report. if so, let em know before 10 am

**Comment by GCrispino** [comment]
@ApertureTS that would be good, if you can.

**Comment by Perryo** [comment]
@ApertureTS @SreeniASU @GCrispino 
Who can bring a laptop for the demo? Mine does not have the openCV package installed.

**Comment by ApertureTS** [comment]
I can bring mine... But I haven't tested it on my laptop

**Comment by Perryo** [comment]
@ApertureTS @GCrispino 
Thats fine, make sure you test before though and make sure it works. Gabriel, we could use your laptop possibly? I know you have it set up.

**Comment by GCrispino** [comment]
@Perryo I can bring my laptop. However, I have not tested all the tasks on it, and also I don't have the extracted features generated from the specific input videos for this task. Does someone have the time to do this(or already did)? I have two classes right now.

**Comment by ApertureTS** [comment]
I'm trying to, but I also have a class, and just installed the matlab lib

**Comment by GCrispino** [comment]
@Perryo @jakepruitt @ApertureTS @SreeniASU 
guys, I tried to run everything to output the features and pushed it to the rep. On the dct and dwt scripts, I wasn't able to run with a n value of 100, it gets an error. Please check the outputs to see if everything is ok.

**Comment by Perryo** [comment]
@GCrispino 
Thanks for taking care of that Gabriel
@SreeniASU 
Have you heard anything back from Yash on what he means by running it with n = 100?

**Comment by SreeniASU** [comment]
I didnt get a reply from Yash. But I talked with one of my friends who
finished his demo just now and what he said was, for DCT and DWT where n =
100 is not possible prof asked them to use n = 64, which is the maximum
value possible.

On Wed, Dec 2, 2015 at 12:07 PM, James notifications@github.com wrote:

> @GCrispino https://github.com/GCrispino
> Thanks for taking care of that Gabriel
> @SreeniASU https://github.com/SreeniASU
> Have you heard anything back from Yash on what he means by running it with
> n = 100?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/SreeniASU/multimedia-information-systems/issues/41#issuecomment-161400520
> .

## 

Sreenivas P Shenoy
M.S. Computer Science & Engineering
Arizona State University,
spsheno1@asu.edu
Cell 480-738-3280

**Comment by Perryo** [comment]
@SreeniASU 
Awesome, sounds good.

**Comment by ApertureTS** [comment]
What time did we sign up for?

**Comment by Perryo** [comment]
@ApertureTS 

> Great, signed up for the 2:15-3:15 block!

This is in the normal classroom, correct?

**Comment by ApertureTS** [comment]
Should be byeng 580. 5th floor

@ApertureTS https://github.com/ApertureTS

Great, signed up for the 2:15-3:15 block!

This is in the normal classroom, correct?

**Comment by Perryo** [comment]
Im here now I don't see anyone

**Comment by ApertureTS** [comment]
Sreeni and I will be there in a few minutes we are in the computer lab on
the 2nd floor

**Comment by Perryo** [comment]
It's 598 come up he's waiting

**Comment by Perryo** [comment]
@jakepruitt 
Just wanted to let you know the presentation went very smoothly. Professor only had one concern about us calculating min and max values for quantization at the block level instead of at the video level. However we explained to him this was an artifact of the assumption we made and verified with Yash and he seemed okay with that. The results also demonstrated to him that the function still worked well regardless. Overall he said we did a very good job and he thought it was good work, so it seems like we nailed this one.
